### PR TITLE
Support accessing native context menu by holding shift key on opening

### LINKED ISSUE #47: Does not respect shift-right-click browser's context menu
_From @juniperbelmont on December 13, 2016 23:27_

### Description

There is an accessibility/usability function in browsers where shift right clicking will open the browser's context menu instead of the custom context menu.

For example, go to Google Drive/Docs/etc. and right click and you'll see the custom context menu, but if you shift right click, you'll see the browser's context menu.

shift-right-click-example

### Expected outcome
Browser's context menu should always come up when shift is held.

### Actual outcome
Only the vaadin-context-menu comes up.

### Live Demo
Any of the demos here:
https://cdn.vaadin.com/vaadin-core-elements/master/vaadin-context-menu/

### Browsers Affected

- [X] Chrome
- [X] Firefox
- [X] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [ ] Android Chrome

_Copied from original issue: vaadin/vaadin-combo-box#360_

### PR DESCRIPTION
**Pull request by platosha**
Fixes #47

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-context-menu/57)

### CODE REVIEW

File: vaadin-contextmenu-event.html [hunk 1]
```diff
@@ -43,7 +43,7 @@
         this.info.touchJob = Polymer.Debounce(this.info.touchJob, function() {
           var t = Polymer.Gestures.findOriginalTarget(e);
           var ct = e.changedTouches[0];
-          if (this.info._ios) {
+          if (this.info._ios && !e.shiftKey) {
```
Comment (limonte) [inline vaadin-contextmenu-event.html]: is `shiftKey` applicable for touch events?

Comment (platosha) [inline vaadin-contextmenu-event.html]: @limonte it’s in the spec: https://www.w3.org/TR/touch-events/#attributes-2

Comment (limonte) [inline vaadin-contextmenu-event.html]: Interesting, thanks for the link. I don't believe it's used in real life https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/shiftKey but still you're technically correct here.

**Comment by manolo** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57#-:-KZQdUJZPKjWsE3meL1u:bnfp4nl)*

**Comment by manolo** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57)*

**Comment by tomivirkki** [comment]
*[vaadin-contextmenu-event.html, line 62 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57#-KZQe4xepDaa2I5yZFVZ:-KZQe4xepDaa2I5yZFV_:b-1cp823) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/84943b9eb57db63b9e5e0ac2e6a9d7802a9c3eb7/vaadin-contextmenu-event.html#L62)):*
> ```HTML
> 
>       contextmenu: function(e) {
>         if (!e.shiftKey) {
> ```

This fixes the issue with our own gesture only. Is that ok? (if you change `open-on="contextmenu"` the fix doesn't apply)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57)*

**Comment by platosha** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-contextmenu-event.html, line 62 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57#-KZQe4xepDaa2I5yZFVZ:-KZQeVNQcalYlJ84LoAu:bz693pj) ([raw file](https://github.com/vaadin/vaadin-context-menu/blob/84943b9eb57db63b9e5e0ac2e6a9d7802a9c3eb7/vaadin-contextmenu-event.html#L62)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This fixes the issue with our own gesture only. Is that ok? (if you change `open-on="contextmenu"` the fix doesn't apply)
</blockquote></details>

Right. On the other hand, I believe, we should not apply this shift key check for any random mouse-/touch-/whatever-event that user wants to open the context menu, as it might result in unexpected behavior on some cases.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57)*

File: vaadin-contextmenu-event.html (hunk 1 above)
Comment (limonte) [inline vaadin-contextmenu-event.html]: LGTM

**Comment by Saulis** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-context-menu/57#-:-KZRC2GoEGzXJHN3rO9L:bnfp4nl)*

**Comment by juniperbelmont** [comment]
@platosha Could we also add this to the docs somewhere? To tell users that this is the expected behaviour?

**Pull request merged**